Add json viewer and update content layout

Kaveh-ap opened the pull request:
### Summary
Add a `BodyViewer` component for displaying request/response bodies with JSON tree viewing and text fallback, plus initial expansion controls throughout `RequestDetails`.

### Motivation
- Improve readability of large JSON payloads with a collapsible tree.
- Provide better defaults via initial expansion controls for sections (headers and bodies).

### Changes
- **New component**: `BodyViewer`
  - Detects JSON strings and renders a collapsible tree; falls back to text view.
  - iOS uses read-only `TextInput` for large content rendering reliability.
  - Supports `data` prop to bypass parsing and render prepared JSON.
  - Basic styling and nested scroll support for better UX.
- **RequestDetails enhancements**
  - New props:
    - `initialRequestHeadersExpanded?: boolean` (default: `true`)
    - `initialResponseHeadersExpanded?: boolean` (default: `true`)
    - `initialRequestBodyExpanded?: boolean` (default: `true`)
    - `initialResponseBodyExpanded?: boolean` (default: `true`)
  - Uses `BodyViewer` for request/response bodies with collapsible sections.
  - Adds “Share full request” and “Share as cURL” actions.
- **Header**
  - Indicator `▼/▶` shows only when `collapsible`.

### Usage
```tsx
<RequestDetails
  request={request}
  onClose={close}
  initialRequestHeadersExpanded={true}
  initialResponseHeadersExpanded={false}
  initialRequestBodyExpanded={true}
  initialResponseBodyExpanded={false}
/>
```

If you want to directly render JSON data:
```tsx
<BodyViewer data={{ foo: 'bar', list: [1, 2, 3] }} initiallyExpanded />
```

### UI/UX
- JSON bodies: Collapsible, indented view with alternating backgrounds per level.
- Text bodies: Scrollable view; on iOS uses read-only `TextInput` for large content stability.

### Backward Compatibility
- Defaults keep previous behavior (all sections expanded initially).
- No breaking API changes.

### Testing
- All tests pass.
- Manual checks for:
  - Collapsible behavior for headers and bodies.
  - JSON parse vs. plain text fallback.
  - Share actions.

### Performance
- JSON parsing only when `content` is a string and `data` isn’t provided.
- Tree rendering uses toggling to limit on-screen content.

### Screenshots/GIFs
<p align="center">
  <img src="https://github.com/user-attachments/assets/ddca4dad-2f17-4544-a515-19202b835091" alt="Screenshot 1" width="45%" />
  <img src="https://github.com/user-attachments/assets/790b5ad3-02c3-424e-a9f7-4f320a9e8977" alt="Screenshot 2" width="45%" />
</p>
<p align="center">
  <img src="https://github.com/user-attachments/assets/09ced7e5-62eb-4fa2-94f6-c49fa8b7563f" alt="Screenshot 3" width="45%" />
  <img src="https://github.com/user-attachments/assets/914ca8fd-87e6-4d4f-89db-41682b0fcf90" alt="Screenshot 4" width="45%" />
</p>

Commit:
feat: Enhance Header and RequestDetails components with collapsible functionality

- Updated Header component to support collapsible behavior with toggle functionality.
- Modified RequestDetails component to utilize collapsible headers for request and response bodies.
- Adjusted styles for better presentation and usability.
- Introduced compact prop in RequestDetails for improved layout control.

Commit:
feat: Introduce BodyViewer component and initial expansion controls for RequestDetails

- Added BodyViewer component to handle JSON and text content display with collapsible functionality.
- Enhanced RequestDetails component to support initial expansion states for request and response headers and bodies.
- Updated README to document new features and usage examples for initial expansion controls.

seyedmostafahasani commented:
@alexbrazier 
Would you please check this PR?
I think it is very useful.

@Kaveh-ap 
Thanks for creating this PR.

Shubham1Jain commented:
@Kaveh-ap Can we also add a search bar to search text in response body, it will help a lot to search for specific things in large payload or response

alexbrazier committed:
Merge branch 'master' into feat/add-json-viewer-and-update-content-layout

alexbrazier commented:
@codex review

Commit:
refactor: Simplify Header component structure and improve collapsible behavior

- Introduced a titleContent variable to streamline rendering of header text.
- Enhanced collapsible functionality by ensuring onToggle is always called when the header is pressed.
- Improved accessibility roles and test IDs for better testing and usability.
